backend/dataall/core/groups/api/resolvers.py

```diff
@@ -71,3 +72,13 @@ def list_cognito_groups(context, source, filter: dict = None):
         if group not in invited_group_uris:
             res.append({"groupName": group})
     return res
+
+
```

Comment:
Clarification: this is needed to setGroups in useGroups. One of the most used hooks. It is called by the user to get its groups. Because it should only allow the user to get ist own groups, I think we should add some check that makes sure that the userid used in the call is the context user id. Without the server-side check, anyone with a JWT token and the API url can call this API call for any user

Reply:
That's another great catch. You are right, this check should be done. 

Option1.
The user_id is not available in the context. In order to do this change, an additional field will be added, to the context. 
In case of external IDP, the user_id will be loaded in this context. This then can be used to check the user_id in the resolver by checking if the context.user_id == user_id. 
In case of Cognito IDP, the user_id will be same as username ( which is the email field ) in the context. 

Option2.
Another way of doing this check is , in the `api_handler.py` file, only for this specific resolver, the query params can be parsed and then the user_id from the `event` can be checked with the query params. 

I personally like option 1, but would like to know what option you find would be the best

Reply:
I prefer option 1 too. Option 2 is too particular for this API call. At the end we will have a context with the user_id. And we will check the same way that we check other permissions against the username

```
class RequestContext:
    """Contains API for every graphql request"""
    db_engine: Engine
    username: str
    groups: List[str]
    user_id: str
```